lib/vrl/compiler/src/vm/machine.rs

```diff
@@ -0,0 +1,533 @@
+    NotEqual,
+
+    /// Pops the top two elements from the stack, pushes a boolean if the two elements are equal.
+    Equal,
+
+    /// Pops the top element from the stack, discarding it.
+    Pop,
+
+    /// Clears the error state from the VM.
+    ClearError,
+
+    /// If the top element of the stack is false advances the instruction pointer by the amount
+    /// set by the ensuing primitive instruction.
+    JumpIfFalse,
+
+    /// If the top element of the stack is true advances the instruction pointer by the amount
+    /// set by the ensuing primitive instruction.
+    JumpIfTrue,
+
+    /// If the error field of the VM is not set advances the instruction pointer by the amount
+    /// set by the ensuing primitive instruction.
+    JumpIfNotErr,
+
+    /// Unconditionally advances the instruction pointer by the amount set by the ensuing
+    /// primitive instruction.
+    Jump,
+
+    /// Takes the path indicated by the ensuing primitive and sets this path with the value
+    /// at the top of the stack. The value is not removed from the stack so it can continue
+    /// to be used.
+    SetPath,
+
+    /// Sets either a success or error path. The next primitive is the index to the target for
+    /// the success path, the one after is the error path. After that is a pointer to a
+    /// constant indicating the default value to set to the success path should there be an error.
+    SetPathInfallible,
+
+    /// Takes the ensuing primitive as a pointer to a path. Retrieves this value from the state
+    /// and pushes this onto the stack.
+    GetPath,
+
+    /// Calls the function indicated by the ensuing primitive.
+    Call,
+
+    /// Creates an array. The ensuing primitive indicates the number of elements in the array.
+    /// This amount of values are popped from the stack. The resulting array is then pushed back on
+    /// the stack.
+    CreateArray,
+
+    /// Creates an object. The ensuing primitive indicatos the number of elements in the object.
+    /// This amount of keys (as string constants) and values are popped from the stack. The resulting object
+    /// is then pushed back on the stack.
+    CreateObject,
+
+    /// Pushes an empty parameter onto the parameter stack. An empty parameter will be a missing optional parameter
+    /// that will be passed to the next function called.
+    EmptyParameter,
+
+    /// Moves a value from the stack onto the parameter stack in preparation for calling a function.
+    MoveParameter,
+
+    /// Moves a static parameter indicated by the ensuing primitive onto the parameter stack. A static parameter
+    /// will have been created by the `compile_argument` function of the `Function` that is about to be called
+    /// at compile time. (Used, for example, to precompile and store regexes at compile time.)
+    MoveStaticParameter,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Instruction {
+    OpCode(OpCode),
+    Primitive(usize),
```

Comment:
It seems like there are several different words they are used interchangeably that all appear to mean the same thing? It might make it easier to understand if they all use the same one.

In the Instruction enum, this is called a `Primitive`, which is an index into the `values` Vec, which hold constants. 

I think it would make more sense if the enum variant was `ConstantIndex`, and the `values` list was renamed to `constants`